websub-ballerina/commons.bal

```diff
@@ -145,253 +83,13 @@ public type Acknowledgement record {
-    }
-
-    # Retrieves the request payload as a `byte[]`.
-    # ```ballerina
-    # byte[]|error payload = notification.getBinaryPayload();
-    # ```
-    #
-    # + return - The message payload as a `byte[]` or else an `error` in case of errors
-    public isolated function getBinaryPayload() returns @tainted byte[]|error {
-        return self.request.getBinaryPayload();
-    }
-
-    # Retrieves the form parameters from the content delivery request as a `map`.
-    # ```ballerina
-    # map<string>|error result = notification.getFormParams();
-    # ```
-    #
-    # + return - The form params as a `map` or else an `error` in case of errors
-    public isolated function getFormParams() returns @tainted map<string>|error {
-        return self.request.getFormParams();
-    }
-}
-
-# Retrieves hub and topic URLs from the `http:response` from a publisher to a discovery request.
-#
-# + response - An `http:Response` received
-# + return - A `(topic, hubs)` if parsing and extraction is successful or else an `error` if not
-public function extractTopicAndHubUrls(http:Response response) returns @tainted [string, string[]]|error {
-    string[] linkHeaders = [];
-    if (response.hasHeader("Link")) {
-        linkHeaders = check response.getHeaders("Link");
-    }
-    
-    if (response.statusCode == http:STATUS_NOT_ACCEPTABLE) {
-        return error WebSubError("Content negotiation failed.Accept and/or Accept-Language headers mismatch");
-    }
-    
-    if (linkHeaders.length() == 0) {
-        return error WebSubError("Link header unavailable in discovery response");
-    }
-
-    int hubIndex = 0;
-    string[] hubs = [];
-    string topic = "";
-    string[] linkHeaderConstituents = [];
-    if (linkHeaders.length() == 1) {
-        linkHeaderConstituents = regex:split(linkHeaders[0], ",");
-    } else {
-        linkHeaderConstituents = linkHeaders;
-    }
-
-    foreach var link in linkHeaderConstituents {
-        string[] linkConstituents = regex:split(link, ";");
-        if (linkConstituents[1] != "") {
-            string url = linkConstituents[0].trim();
-            url = regex:replaceAll(url, "<", "");
-            url = regex:replaceAll(url, ">", "");
-            if (strings:includes(linkConstituents[1], "rel=\"hub\"")) {
-                hubs[hubIndex] = url;
-                hubIndex += 1;
-            } else if (strings:includes(linkConstituents[1], "rel=\"self\"")) {
-                if (topic != "") {
-                    return error WebSubError("Link Header contains > 1 self URLs");
-                } else {
-                    topic = url;
-                }
-            }
-        }
-    }
-
-    if (hubs.length() > 0 && topic != "") {
-        return [topic, hubs];
-    }
-    return error WebSubError("Hub and/or Topic URL(s) not identified in link header of discovery response");
-}
+public type RequestQueryParams record {|
+    string hubMode = "";
+    string hubTopic = "";
+    string hubChallenge = "";
```

Comment:
Above doesn't look like ballerina way of coding :) Any reason why we don't make those `string?` ?

Reply:
* Updated the `RequestQueryParams` type.